core/src/main/java/org/elasticsearch/index/shard/StoreRecovery.java

```diff
@@ -114,6 +115,9 @@ boolean recoverFromLocalShards(BiConsumer<String, MappingMetaData> mappingUpdate
             indexShard.mapperService().merge(indexMetaData, MapperService.MergeReason.MAPPING_RECOVERY, true);
             // now that the mapping is merged we can validate the index sort configuration.
             Sort indexSort = indexShard.getIndexSort();
+            final boolean isSplit = indexMetaData.getNumberOfShards() < indexShard.indexSettings().getNumberOfShards();
```

Comment:
can we call `indexMetaData` `sourceMetaData` - this would save some head scratching :)